### PR TITLE
Add Authorization rules

### PR DESCRIPTION
**Pull request by butlermuwo**
## Add authorization rules.
In this pull request we did the following:
- Installed CanCanCan in our project.
- Added a role column to the users table using migration.
- A user can delete a post if it is theirs or if they have an admin role.
- A user can delete a comment if it is theirs or if they have an admin role

### CODE REVIEW

File: app/views/posts/show.html.erb [hunk 1]
```diff
@@ -11,6 +12,7 @@
 <% @post.recent_comments.each do |comment| %>
     <div class="comment">
         <p><%= comment.text %></p>
+        <%= button_to 'Delete comment', post_comment_path(post_id: @post.id, comment_id: comment.id), method: :delete %>
```
Comment (medaminedev66) [inline app/views/posts/show.html.erb]: - [x] Following the project requirements, a user can delete a comment if it is theirs or if they have an admin role. I still can delete the comments of other users. Please make sure only the author of the comment can delete their comment.

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -15,9 +15,21 @@ def create
     end
```
Comment (medaminedev66) [inline app/controllers/comments_controller.rb:14]: - [x] When I log in using another account and I try to add a comment for another's user post, It shows me the following error:

image

- Please make sure that the current user is able to like and add comments to other users' posts.

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @butlermuwo  and @NataliaPoletaeva  😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- The repo looks professional ✔️
- There are no linter errors ✔️
- Cancancan tool has been used for authorization. ✔️

## Required Changes ♻️
- [x] Pull Request should have a descriptive title telling you what changes are inside. Please update that.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (butlermuwo) [reply app/controllers/comments_controller.rb:14]: @medaminedev66  Thanks we have implemented as suggested.

File: app/views/posts/show.html.erb (hunk 1 above)
Comment (butlermuwo) [reply app/views/posts/show.html.erb]: Thanks @medaminedev66  we have implemented as suggested.

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi Team, 

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by butlermuwo** [comment]
Thanks @Mihndim2020

**Pull request merged**